packages/amplify_authenticator/lib/src/widgets/form_fields/sign_in_form_field.dart

```diff
@@ -79,30 +77,241 @@ class SignInFormField extends AuthenticatorFormField {
         throw StateError('$field is not supported as a sign in field');
     }
   }
+}
+
+class SignInPasswordFormField extends SignInFormField {
+  const SignInPasswordFormField({
+    Key? key,
+    FormFieldValidator<String>? validator,
+    bool? required,
+  }) : super._(
+          key: key,
+          field: AuthenticatorFormFieldType.password,
+          titleKey: InputResolverKey.passwordTitle,
+          hintTextKey: InputResolverKey.passwordHint,
+          validator: validator,
+          required: required,
+        );
 
   @override
-  SignInFormFieldState createState() => SignInFormFieldState();
+  _SignInPasswordFormFieldState createState() =>
+      _SignInPasswordFormFieldState();
 }
 
-class SignInFormFieldState extends AuthenticatorFormFieldState {
+class _SignInPasswordFormFieldState extends AuthenticatorFormFieldState
+    with AuthenticatorTextFormFieldMixin {}
+
+class SignInUsernameFormField extends SignInFormField {
+  const SignInUsernameFormField({
+    Key? key,
+    FormFieldValidator<String>? validator,
+    bool? required,
+  }) : super._(
+          key: key,
+          field: AuthenticatorFormFieldType.username,
+          titleKey: InputResolverKey.usernameTitle,
+          hintTextKey: InputResolverKey.usernameHint,
+          validator: validator,
+          required: required,
+        );
+
   @override
-  Widget build(BuildContext context) {
+  _SignInUsernameFormFieldState createState() =>
+      _SignInUsernameFormFieldState();
+}
+
+class _SignInUsernameFormFieldState extends AuthenticatorFormFieldState
+    with AuthenticatorUsernameFormFieldMixin {}
+
+// TODO: Move AuthenticatorTextFormFieldMixin to it's own file
```

Comment:
mixins would move back to their own file if this approach were used for each form. I just put them here temporarily